### PR TITLE
BigTable: Add admin app profile methods on Instance

### PR DESCRIPTION
**Pull request by aneepct**
Add methods to the Instance class to manage app profiles:
1. create_app_profile
2. get_app_profile
3. list_app_profiles
4. update_app_profile
5. delete_app_profile

### CODE REVIEW

File: bigtable/google/cloud/bigtable/instance.py [hunk 1]
```diff
@@ -239,3 +244,219 @@ def list_tables(self):
...
+        :type: description: str
+        :param: description: (Optional) Long form description of the use
+                                case for this AppProfile.
+
+        :type: ignore_warnings: bool
+        :param: ignore_warnings: (Optional) If true, ignore safety checks when
+                                    creating the app profile.
+
+        :type: metadata: [Sequence[Tuple[str, str]]]
+        :param: metadata: (Optional) Additional metadata that is provided to
```
Comment (sduskis) [inline bigtable/google/cloud/bigtable/instance.py]: I'm not sure what metadata is.  There isn't any metadata in the [protobuf definition](https://github.com/googleapis/googleapis/blob/master/google/bigtable/admin/v2/bigtable_instance_admin.proto#L439).

If you don't know what this is, please remove it.

File: bigtable/google/cloud/bigtable/instance.py [hunk 2]
```diff
@@ -282,3 +286,189 @@ def list_tables(self):
...
+        """Creates an app profile within an instance.
+
+        :type: app_profile_id: str
+        :param app_profile_id: The unique name for the new app profile.
+
+        :type: routing_policy_type: int
+        :param: routing_policy_type: There are two routing policies
+                                    ROUTING_POLICY_TYPE_ANY = 1 and
+                                    ROUTING_POLICY_TYPE_SINGLE = 2.
+                                    By default it is ROUTING_POLICY_TYPE_ANY
```
Comment (sduskis) [inline bigtable/google/cloud/bigtable/instance.py]: Please update the routing type comment, now that there is no default.

File: bigtable/google/cloud/bigtable/instance.py [hunk 3]
```diff
@@ -282,3 +286,189 @@ def list_tables(self):
...
+
+        :type: update_mask: list
+        :param: update_mask: Name of the parameters of AppProfiles that
+                                needed to update.
+
+        :type: routing_policy_type: int
+        :param: routing_policy_type: There are two routing policies
+                                    ROUTING_POLICY_TYPE_ANY = 1 and
+                                    ROUTING_POLICY_TYPE_SINGLE = 2.
+                                    By default it is ROUTING_POLICY_TYPE_ANY
```
Comment (sduskis) [inline bigtable/google/cloud/bigtable/instance.py]: Please remove the comment about a default routing type.

File: bigtable/google/cloud/bigtable/instance.py [hunk 4]
```diff
@@ -282,3 +286,189 @@ def list_tables(self):
...
+        if routing_policy_type == ROUTING_POLICY_TYPE_SINGLE:
+            single_cluster_routing = (
+                instance_pb2.AppProfile.SingleClusterRouting(
+                    cluster_id=cluster_id,
+                    allow_transactional_writes=allow_transactional_writes
+                ))
+
+        app_profile = instance_pb2.AppProfile(
+            name=name, description=description,
+            multi_cluster_routing_use_any=multi_cluster_routing_use_any,
```
Comment (sduskis) [inline bigtable/google/cloud/bigtable/instance.py]: In Java protobufs, one cannot set both multi and single cluster routing policies at the same time.  Please add a system test for both cases (single and multi cluster).

File: bigtable/tests/unit/test_instance.py [hunk 5]
```diff
@@ -408,6 +408,304 @@ def test_list_tables_failure_name_bad_before(self):
...
+        # Patch the stub used by the API method.
+        client._instance_admin_client = instance_api
+        bigtable_instance_stub = (
+            client._instance_admin_client.bigtable_instance_admin_stub)
+        bigtable_instance_stub.CreateAppProfile.side_effect = [response_pb]
+
+        # Perform the method and check the result.
+        app_profile_id = 'appProfileId1262094415'
+        result = instance.create_app_profile(app_profile_id=app_profile_id,
+                                             routing_policy_type=1)
```
Comment (sduskis) [inline bigtable/tests/unit/test_instance.py]: Can you please add a test to make sure that the request proto you expect is sent to the stub?

File: bigtable/tests/unit/test_instance.py [hunk 6]
```diff
@@ -408,6 +408,304 @@ def test_list_tables_failure_name_bad_before(self):
...
+
+        # Patch the stub used by the API method.
+        client._instance_admin_client = instance_api
+        bigtable_instance_stub = (
+            client._instance_admin_client.bigtable_instance_admin_stub)
+        bigtable_instance_stub.CreateAppProfile.side_effect = [response_pb]
+
+        # Perform the method and check the result.
+        app_profile_id = 'appProfileId1262094415'
+        result = instance.create_app_profile(app_profile_id=app_profile_id,
```
Comment (sduskis) [inline bigtable/tests/unit/test_instance.py]: Can you please add a test to make sure that the request proto you expect is sent to the stub?

File: bigtable/tests/unit/test_instance.py (hunk 5 above)
Comment (aneepct) [reply bigtable/tests/unit/test_instance.py]: @sduskis Working on mock stub calls in unit tests

File: bigtable/tests/unit/test_instance.py (hunk 6 above)
Comment (aneepct) [reply bigtable/tests/unit/test_instance.py]: @sduskis Working on mock stub calls in unit tests

**Pull request merged**